maas-agent/src/charm.py

```diff
@@ -36,6 +38,7 @@
 MAAS_SNAP_CHANNEL = "3.5/stable"
 
 
+@trace_charm(tracing_endpoint="charm_tracing_endpoint")
```

Comment:
What about `extra_types` on the classes we setup for our integrations? `COSAgentProvider`, `MaasRegionProvider`. Did you try to add them and observe any differences in the captured traces?